zerver/models.py

```diff
@@ -492,8 +493,11 @@ def currently_used_upload_space_bytes(self) -> int:
             return 0
         return used_space
 
+    def is_on_limited_plan(self) -> bool:
+        return self.plan_type in [Realm.LIMITED, Realm.SPONSORSHIP_PENDING]
```

Comment:
It's a little awkward to need to change these conditionals; did you think about the data model of just adding a `sponsorship_pending` flag on `Customer` for tracking this state, rather than putting it on Realm?  I feel like we want the behavior to be exactly like LIMITED except on billing/plans pages, and it might be cleaner to arrange that by not modifying `Realm.plan_type`?